src/main/scala/com/madgag/rfc5988link/LinkParser.scala

```diff
@@ -17,25 +17,27 @@
 package com.madgag.rfc5988link
 
 import okhttp3.HttpUrl
-import fastparse.all._
+import fastparse._
+import NoWhitespace._
+
+import com.madgag.scala.collection.decorators._
 
 case class LinkTarget(url: HttpUrl, attributes: Seq[(String, String)]) {
 
-  lazy val attributeMap = attributes.groupBy(_._1).mapValues(_.map(_._2))
+  lazy val attributeMap: Map[String, Seq[String]] = attributes.groupBy(_._1).mapV(_.map(_._2))
 
-  lazy val relOpt = attributeMap.get("rel").flatMap(_.headOption)
+  lazy val relOpt: Option[String] = attributeMap.get("rel").flatMap(_.headOption)
 }
 
 object LinkParser {
-  val url: P[HttpUrl] = P("<" ~ CharsWhile(_ != '>', min = 1).! ~ ">").map(HttpUrl.parse)
+  def url[_: P]: P[HttpUrl] = P("<" ~/ CharsWhile(_ != '>', 1).! ~ ">").map(HttpUrl.parse)
 
-  val linkParam: P[(String, String)] =
-    P("; " ~ CharsWhile(_ != '=', min = 1).! ~ "=\"" ~ CharsWhile(_ != '"', min = 1).! ~ "\"" )
+  def linkParam[_: P]: P[(String, String)] =
+    P("; " ~ CharsWhile(_ != '=',1).! ~ "=\"" ~ CharsWhile(_ != '"',1).! ~ "\"" )
 
-  val linkTarget: P[LinkTarget] = (url ~ linkParam.rep).map {
+  def linkTarget[_: P]: P[LinkTarget] = (url ~ linkParam.rep).map {
     case (a, b) => LinkTarget(a, b)
   }
 
-  val linkValues: P[Seq[LinkTarget]] = linkTarget.rep(sep = ", ")
-
+  def linkValues[_: P]: P[Seq[LinkTarget]] = linkTarget.rep(sep = ", ")
```

Comment:
These changes come from the migration from Fastparse v1 to v2:

https://www.lihaoyi.com/post/Fastparse2EvenFasterScalaParserCombinators.html